src/local-settings.js

```diff
@@ -1,62 +1,90 @@
-const DEFAULT_ETH_NODE_KEY = 'ETH_NODE_KEY'
-const ENS_REGISTRY_ADDRESS_KEY = 'ENS_REGISTRY_ADDRESS_KEY'
-const ETH_NETWORK_TYPE_KEY = 'ETH_NETWORK_TYPE_KEY'
-const IPFS_GATEWAY_KEY = 'IPFS_GATEWAY_KEY'
-const IPFS_RPC_KEY = 'IPFS_RPC_KEY'
+// List of configurable settings
+const DEFAULT_ETH_NODE = 'DEFAULT_ETH_NODE'
+const ENS_REGISTRY_ADDRESS = 'ENS_REGISTRY_ADDRESS'
+const ETH_NETWORK_TYPE = 'ETH_NETWORK_TYPE'
+const IPFS_GATEWAY = 'IPFS_GATEWAY'
+const IPFS_RPC = 'IPFS_RPC'
 
-// (protocol)://(host):(port)
-const IPFS_RPC_REGEX = /^([\w]*):\/\/(.*):(\d*)$/
+const CONFIGURATION_KEYS = [
+  DEFAULT_ETH_NODE,
+  ENS_REGISTRY_ADDRESS,
+  ETH_NETWORK_TYPE,
+  IPFS_GATEWAY,
+  IPFS_RPC,
+].reduce(
+  (acc, option) => ({
+    ...acc,
+    [option]: {
+      storageKey: `${option}_KEY`,
+      // REACT_APP_ prefix required for react-scripts
+      environmentKey: `REACT_APP_${option}`,
+    },
+  }),
+  {}
+)
 
 export function getDefaultEthNode() {
+  const keys = CONFIGURATION_KEYS[DEFAULT_ETH_NODE]
   return (
-    window.localStorage.getItem(DEFAULT_ETH_NODE_KEY) ||
-    process.env.REACT_APP_DEFAULT_ETH_NODE ||
+    window.localStorage.getItem(keys.storageKey) ||
+    process.env[keys.environmentKey] ||
     'ws://rinkeby.aragon.network:8546'
```

Comment:
We could do this a function, passing the key name (`DEFAULT_ETH_NODE` in this case) and the default value (`     'ws://rinkeby.aragon.network:8546'` here), as it's used several times below.

Reply:
See 7e5951a